### ISSUE TITLE
NewComment into Post

### ISSUE THREAD

**Issue by tayfunyasar** [abrudtkuhl/WordPressSharp#46]
How to add comments into the post?

I think its not ready for now. Am i right?

**Comment by KeesCBakker** [comment]
I'm looking into it. Might have something for you in a few hours...

**Comment by KeesCBakker** [comment]
Added it to my own branch and created a pull request. Also added a unit test for it. Seems to work like a charm. So I'll say enjoy @tayfunyasar! The unit test shows how to create a comment.

**Comment by tayfunyasar** [comment]
Thanks @KeesCBakker 

my addComment methods are

public static int addComment(string postId, string content, string author)
    {
      Comment comment = setComment(postId, content, author);
      using (var client = getWordpressClient())
      {
        return client.NewComment(comment);
      }
    }
    public static Comment setComment(string postId, string content, string author)
    {
      return new Comment
      {
        PostId = postId,
        Content = content,
        Author = author 
      };
    }

and im adding comments like this,
addComment("1453","John","Hello");

but i see all comments are written by admin. how to add comment as guest?

**Comment by tayfunyasar** [comment]
Any updates?

**Comment by tayfunyasar** [comment]
Anything new?

**Comment by tayfunyasar** [comment]
Still waiting. Thanks

**Comment by tayfunyasar** [comment]
Its been 11 days. Still waiting answer T_T

**Comment by abrudtkuhl** [comment]
waiting on what? the pull request has been merged that adds this functionality. 

It is a good time to note - this is very much a side project. The fastest way to add functionality is to build it and submit a pull request rather than pinging a comment thread. Otherwise I get back to it about once every two weeks aside from reviewing PR's and merging.

**Comment by tayfunyasar** [comment]
No abrud. I'm waiting for a clue. I cant add comment with a custom "Name", "Email" and "Website"

**Comment by KeesCBakker** [comment]
Any news on this issue?

**Comment by KeesCBakker** [comment]
Looks like WordPress will use the user that is used in the RPC. Still looking into it through through the forums...

**Comment by tayfunyasar** [comment]
I'm still curious about this question. Still cant add comments with custom name & mail

**Comment by KeesCBakker** [comment]
I've added an edit function. But still the user info doesn't get edited. The f

>__Note on Anonymous Comments__
>By default, the username and password parameters are required. However, if you wish to allow anonymous comments, you can use the xmlrpc_allow_anonymous_comments filter:
>
>```add_filter( 'xmlrpc_allow_anonymous_comments', '__return_true' );```
>
>In such cases, the username and password parameters can be left as empty strings.

I have no WordPress installation to test this. You might want to add this filter and try without username / password.

**Comment by tayfunyasar** [comment]
I've edited  my wp-includes/class-wp-xmlrpc-server.php file at line 3207

```$allow_anon = apply_filters('xmlrpc_allow_anonymous_comments', '__return_true');  ```

still writing comments from admin/logged user T_T

my code: https://github.com/abrudtkuhl/WordPressSharp/issues/46#issuecomment-96817665

**Comment by abrudtkuhl** [comment]
You aren't going to be able to post anonymous comments through the xml-rpc API 




Also - you probably already know but you should never edit core

**Comment by tayfunyasar** [comment]
Really? cant i add annon comments with WordpressSharp? its not possible right?..

I will restore class-wp-xmlrpc-server.php today.

**Comment by abrudtkuhl** [comment]
It's not a WordPressSharp thing - it's the XML-RPC that we interface with
WordPress. http://codex.wordpress.org/XML-RPC_WordPress_API

Andy Brudtkuhl
@abrudtkuhl <http://twitter.com/abrudtkuhl>

**Comment by tayfunyasar** [comment]
I've put the code @KeesCBakker shared into my theme's functions.php at first line. But still cant add annonyms comments.. Can you directly tell me what to do about this? isnt it possible?

**Comment by abrudtkuhl** [comment]
No sorry it's definitely outside the scope of this library. Why would you want this other than for spamming? You can always insert directly into database

**Comment by KeesCBakker** [comment]
I would like this feature very very much. So I'm not in agreement with @abrudtkuhl. Yet it is something we cannot control, as WordPress has sealed it. Hopefully we'll be able to do this in a future version.

**Comment by tayfunyasar** [comment]
I've tried to connect my mysql db using c#.net. I asked that to my hosting company, they said "you are in shared hosting, its not allowed." In that case, we need that feature asap :+1: I'm waiting your exciting news about that @KeesCBakker ! 

BTW thats the only library we can do something in wordpress & got a lof of helper developers. We are The One^^

**Comment by tayfunyasar** [comment]
is there any updates? :)

**Comment by abrudtkuhl** [comment]
There's no way to do it with the WordPressSharp library as it's reliant on the xml-rpc API in the WordPress core




The only way to do it is directly accessing the db or writing something in a custom plugin to do the insertions using the WP API. Either way it's not in scope with this project
